File: android/app/src/main/java/com/lightningkite/khrysalistemplate/layouts/SelectDemoXml.kt
//
// SelectDemoXml.swift
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysalistemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalistemplate.R

class SelectDemoXml {

    lateinit var list: androidx.recyclerview.widget.RecyclerView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.select_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        list = view.findViewById<androidx.recyclerview.widget.RecyclerView>(R.id.list)
        
        
        return view
    }
}

File: web/src/layout/SelectDemoXml.ts
//
// SelectDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './select_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.SelectDemoXml
export class SelectDemoXml {
    xmlRoot!: HTMLElement
    list!: HTMLDivElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.list = getViewById<HTMLDivElement>(view, "list")
        return view
    }
}
